pkg/pushsync/metrics.go

```diff
@@ -14,48 +14,134 @@ type metrics struct {
+			Help:      "Total chunks sent.",
+		}),
+		ChunksReceivedCounter: prometheus.NewCounter(prometheus.CounterOpts{
 			Namespace: m.Namespace,
 			Subsystem: subsystem,
-			Name:      "send_chunk_time_taken",
-			Help:      "Total time taken to send a chunk.",
+			Name:      "received_chunk",
+			Help:      "Total chunks received.",
 		}),
 		SendChunkErrorCounter: prometheus.NewCounter(prometheus.CounterOpts{
 			Namespace: m.Namespace,
 			Subsystem: subsystem,
 			Name:      "send_chunk_error",
 			Help:      "Total no of time error received while sending chunk.",
 		}),
-		MarkAndSweepTimer: prometheus.NewCounter(prometheus.CounterOpts{
+		ReceivedChunkErrorCounter: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: m.Namespace,
+			Subsystem: subsystem,
+			Name:      "received_chunk_error",
+			Help:      "Total no of time error received while receiving chunk.",
+		}),
+		ReceiptsReceivedCounter: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: m.Namespace,
+			Subsystem: subsystem,
+			Name:      "received_receipts",
+			Help:      "Total no of times receipts received.",
+		}),
+		ReceiptsSentCounter: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: m.Namespace,
+			Subsystem: subsystem,
+			Name:      "sent_receipts",
+			Help:      "Total no of times receipts are sent.",
+		}),
+		SendReceiptErrorCounter: prometheus.NewCounter(prometheus.CounterOpts{
 			Namespace: m.Namespace,
 			Subsystem: subsystem,
-			Name:      "mark_and_sweep_time",
-			Help:      "Total time spent in mark and sweep.",
+			Name:      "sent_receipts_error",
+			Help:      "Total no of times receipts were sent and error was encountered.",
+		}),
+		ReceiveReceiptErrorCounter: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: m.Namespace,
+			Subsystem: subsystem,
+			Name:      "receive_receipt_error",
+			Help:      "Total no of time error received while receiving receipt.",
 		}),
 
-		ChunksInBatch: prometheus.NewGauge(prometheus.GaugeOpts{
+		ErrorSettingChunkToSynced: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: m.Namespace,
+			Subsystem: subsystem,
+			Name:      "cannot_set_chunk_sync_in_DB",
+			Help:      "Total no of times the chunk cannot be synced in DB.",
+		}),
+		RetriesExhaustedCounter: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: m.Namespace,
+			Subsystem: subsystem,
+			Name:      "chunk_retries_exhausted",
+			Help:      "CHunk retries exhausted.",
+		}),
+		InvalidReceiptReceived: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: m.Namespace,
+			Subsystem: subsystem,
+			Name:      "invalid_receipt_receipt",
+			Help:      "Invalid receipt received from peer.",
+		}),
+		SendChunkTimer: prometheus.NewHistogram(prometheus.HistogramOpts{
+			Namespace: m.Namespace,
+			Subsystem: subsystem,
+			Name:      "send_chunk_time_histogram",
+			Help:      "Histogram for Time taken to send a chunk.",
+			Buckets:   []float64{0.01, 0.1, 0.25, 0.5, 1, 2.5, 5, 10},
```

Comment:
Thanks for adding histograms @jmozah. I assume that you reused the same bucket distribution as in the ResponseDuration metric. I think that we can adjust that here as less than `0.01s` is not likely to happen, but more than `10s` is possible. Would it better to have a distribution from `0.1s` to `60s`, I think that this would be more useful.